x/tokenfactory/keeper/admins_test.go

```diff
@@ -185,82 +185,76 @@ func (suite *KeeperTestSuite) TestBurnDenom() {
 }
 
 func (suite *KeeperTestSuite) TestChangeAdminDenom() {
-	var (
-		denom    string
-		addr0bal int64
-		addr1bal int64
-	)
-
 	for _, tc := range []struct {
-		desc  string
-		setup func()
-		valid bool
+		desc               string
+		msgChangeAdmin     func(denom string)
+		expectedAdminIndex int
+		msgMint            func(denom string)
```

Comment:
Lets make the functions returns the message type directly?

Then LGTM, ty for doing this

Reply:
The reason I used a separate function is bc we need to take in the changed denom as a parameter of a function to used the updated denom. 

A possible option perhaps is changing the field to become `msgChangeAdmin func(denom string) *types.MsgChangeAdmin`, where the function directly returns the message?

Reply:
yeah!

Reply:
yeah agreed that testing layers should be all together, not in separate fields!